ts/components/ZendeskChatComponent.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { useEffect } from "react";
+import * as pot from "italia-ts-commons/lib/pot";
+import { fromNullable, Option } from "fp-ts/lib/Option";
+import { GlobalState } from "../store/reducers/types";
+import { isLoggedInWithSessionInfo } from "../store/reducers/authentication";
+import { Dispatch } from "../store/actions/types";
+import {
+  AnonymousIdentity,
+  initSupportAssistance,
+  JwtIdentity,
+  openSupportTicket,
+  setUserIdentity,
+  showSupportTickets,
+  ZendeskConfig,
+  zendeskDefaultAnonymousConfig,
+  zendeskDefaultJwtConfig
+} from "../utils/supportAssistance";
+import { profileSelector } from "../store/reducers/profile";
+import { InitializedProfile } from "../../definitions/backend/InitializedProfile";
+import IconFont from "./ui/IconFont";
+import { IOColors } from "./core/variables/IOColors";
+import ButtonDefaultOpacity from "./ButtonDefaultOpacity";
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+const ZendeskChatComponent: React.FC<Props> = (props: Props) => {
```

Comment:
```suggestion
const ZendeskChatComponent = () => {
  const zendeskToken = useIOSelector(zendDeskTokenSelector);
  const profile = useIOSelector(profileSelector);
```
I guess a refactoring could improve the component readability. I suggest the usage of redux hooks

In the suggested scenario we could remove this block
```typescript
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ZendeskChatComponent);
```

within the definition of `mapStateToProps` and `mapDispatchToProps`

Note I included a new selector `zendDeskTokenSelector` - `ts/store/reducers/authentication.ts`
```typescript
export const zendDeskTokenSelector = (state: GlobalState): string | undefined =>
  isLoggedInWithSessionInfo(state.authentication)
    ? state.authentication.sessionInfo.zendeskToken
    : undefined;
```

we should always use the selector pattern to access the state to avoid we can extract the same data in different point and in different way around the code base

Reply:
Thank you for the suggestion!
Applied in [0ca5c2e](https://github.com/pagopa/io-app/pull/3508/commits/0ca5c2ef5cdb4597ea8693728190679bcd195a7e) and in  [3d041b2](https://github.com/pagopa/io-app/pull/3508/commits/3d041b2c0ed947ee919f2bb530e9d58f4192e224)